src/Controls/StreetControls.js

```diff
@@ -0,0 +1,111 @@
+    root.camera.position.lerpVectors(root.positionFrom, root.positionTo, dampingProgress);
+}
+/**
+ * @classdesc Camera controls that can follow a path.
+ * It is used to simulate a street view.
+ * It stores a currentPosition and nextPosition, and do a camera traveling to go to next position.
+ *
+ * <ul> Bindings inherited from FirstPersonControls
+ * <li><b> up + down keys : </b> forward/backward </li>
+ * <li><b> left + right keys: </b> strafing movements </li>
+ * <li><b> pageUp + pageDown: </b> vertical movements </li>
+ * <li><b> mouse click+drag: </b> pitch and yaw movements (as looking at a panorama) </li>
+ * </ul>
+ * <ul> Bindings added
+ * <li><b> keys Z : </b> Move camera to the next position </li>
+ * </ul>
+ * @extends FirstPersonControls
+ */
+class ImmersiveControls extends FirstPersonControls {
+    /**
+     * @constructor
+     * @param { View } view - View where this control will be used
+     * @param { Object } options - Configuration of this controls
+     * @param { number } options.animationDuration - Duration of the animation to another panoramic.
+     */
+    constructor(view, options = {}) {
+        super(view, options);
+
+        this.isStreetControls = true;
+
+        // store layers
+        this.layers = [];
+        this.currentLayerIndex = 0;
+
+        // manage camera movements
+        this.player = new AnimationPlayer();
+        this.animationMoveCamera = new AnimatedExpression({
+            duration: options.animationDuration || 50,
+            root: this,
+            expression: moveCameraExp,
+            name: 'Move camera',
+        });
+        this.player.addEventListener('animation-frame', this.updateView.bind(this));
+
+        // twos positions used by this control : current and next
+        this.currentPosition = undefined;
+        this.nextPosition = undefined;
+    }
+
+    setCurrentPosition(newCurrentPosition) {
+        this.currentPosition = newCurrentPosition;
+    }
+
+    setNextPosition(newNextPosition) {
+        this.nextPosition = newNextPosition;
+    }
+
+    /**
+     * Sets the camera to the current position, looking at the next position.
+     */
+    setCameraToCurrentPosition() {
+        this.setCameraOnPosition(this.currentPosition, this.nextPosition);
+    }
+
+    /**
+     * Set the camera on a position, looking at another position.
+     *
+     * @param      { THREE.Vector3 }  currentPosition   The position to set the camera
+     * @param      { THREE.Vector3 }  nextPosition      The position where the camera look at.
+     */
+    setCameraOnPosition(currentPosition, nextPosition) {
+        if (!currentPosition || !this.nextPosition) {
+            return;
+        }
+        this.camera.position.copy(currentPosition);
+        this.camera.up.copy(currentPosition.clone().multiplyScalar(1.1).normalize());
+        this.camera.lookAt(nextPosition.clone());
```

Comment:
it'sn't necessary to clone.

Reply:
I have an old example of ImmersiveView in planarView, I can add it on this PR to verify that it works well on planar , and eventually debug it. Should I add it on this PR ? WDYT ?

Reply:
if you want but it's not necessary at the moment